frontend/server/api/auth/[...].ts

```diff
@@ -0,0 +1,81 @@
+// file: ~/server/api/auth/[...].ts
+import CredentialsProvider from 'next-auth/providers/credentials'
+import { NuxtAuthHandler } from '#auth'
+
+
+async function getCustomer(email:string) {
+  return await $fetch('/api/auth/customer', {
+    method: 'POST',
+    body: {
+      email: email,
+    },
+  });
+}
+
+export default NuxtAuthHandler({
+  providers: [
+    // @ts-expect-error You need to use .default here for it to work during SSR. May be fixed via Vite at some point\
+    // ^ from sidebase docs
+    CredentialsProvider.default({
+      name: 'Credentials',
+      async authorize(credentials: any) {
+        const customer = await getCustomer(credentials.email);
+        let user = null;
+        //@ts-expect-error
+        if (credentials.email === customer.email && credentials.password == "password") {
+          user = {
+            //@ts-expect-error
+            email:customer.email,
+            //@ts-expect-error
+            id: customer.id,
+            //@ts-expect-error
+            firstName: customer.firstName,
+            //@ts-expect-error
+            lastName: customer.lastName,
+            //@ts-expect-error
```

Comment:
Can you remove these `@ts-expect-error` annotations? Use or create an interface for this `User` object.

Reply:
Resolved